sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -182,9 +180,10 @@ def _get_account(self, username=None, tenant_id=None):
         raise CredentialUnavailableError(message=message)
 
     def _get_refresh_tokens(self, account):
-        return self._cache.find(
+        cache_entries = self._cache.find(
             TokenCache.CredentialType.REFRESH_TOKEN, query={"home_account_id": account.get("home_account_id")}
         )
+        return (token["secret"] for token in cache_entries if "secret" in token)
```

Comment:
Do we need return None?

Reply:
No. Callers of this method expect an iterable.

Reply:
So what if cache_entries is None?

Reply:
That would be an MSAL bug (`find` returns a list).